### ISSUE TITLE
Unexpected result from pre-trained model

### ISSUE THREAD

**Issue by cv38** [minar09/cp-vton-plus#10]
I used the pretrained model and resulting images have a black face and I am getting unexpected result.
[
![000501_0](https://user-images.githubusercontent.com/66823829/89190569-45847580-d5bf-11ea-98aa-0c05a24203a0.jpg)
](url)

Can you please help

**Comment by minar09** [comment]
Hi @cv38 , can you please provide your environment details, and the testing procedure step-by-step as you did? This is a very unique issue raised so far. Thanks.

**Comment by cv38** [comment]
I have a windows 10 laptop with gtx2060 gpu. I follwed the steps provided in readme.
1. Downloaded the code and data
2. python dataset_neck_skin_correction.py,
3.python body_binary_masking.py

Downloaded checkpoints
python test.py --name GMM --stage GMM --workers 4 --datamode test --data_list test_pairs.txt --checkpoint checkpoints/GMM/gmm_final.pth
copied all the result folders into data folder

python test.py --name TOM --stage TOM --workers 4 --datamode test --data_list test_pairs.txt --checkpoint checkpoints/TOM/tom_final.pth

**Comment by minar09** [comment]
The sample you provided here, expected result should be like below: 
![000501_0](https://user-images.githubusercontent.com/3684134/89194587-88a41000-d5e2-11ea-80dc-0546d63070ca.jpg)

If you run the default code, there should be other folders besides `try-on`. Can you please check those results, for example, `im_h`, `m_composite`, `p_rendered`, `shape`, and the warped cloth output from GMM? Thank you.

**Comment by cv38** [comment]
![000501_0](https://user-images.githubusercontent.com/66823829/89197235-0a873f80-d5c9-11ea-8fb9-fb7c082731f0.jpg)
![000501_0](https://user-images.githubusercontent.com/66823829/89197261-12df7a80-d5c9-11ea-9cc2-75a0570bf921.jpg)
![000501_0](https://user-images.githubusercontent.com/66823829/89197283-18d55b80-d5c9-11ea-89fb-afec158fdfff.jpg)
![000501_0](https://user-images.githubusercontent.com/66823829/89197296-1ecb3c80-d5c9-11ea-85ad-abc99927cdb1.jpg)

**Comment by cv38** [comment]
these are the images in folder mentioned by you.

**Comment by cv38** [comment]
![000501_0](https://user-images.githubusercontent.com/66823829/89197665-b29d0880-d5c9-11ea-8659-0eca9cd20f18.jpg)
Warped cloth, i have changed the target cloth

**Comment by minar09** [comment]
Hmm, its strange. Looks like other results are also coming out as unexpected,  e.g., im_h, composite map etc.

Did you test with the original pairs? Is it also producing unexpected results?

Can you please check the opencv, pillow versions as well?

It's hard to get just by the results. If you can debug the sample to see the intermediate results,  that would be great.

**Comment by thaithanhtuan** [comment]
> ![000501_0](https://user-images.githubusercontent.com/66823829/89197665-b29d0880-d5c9-11ea-8659-0eca9cd20f18.jpg)
> Warped cloth, i have changed the target cloth

How did you change the target cloth?

**Comment by cv38** [comment]
> > ![000501_0](https://user-images.githubusercontent.com/66823829/89197665-b29d0880-d5c9-11ea-8659-0eca9cd20f18.jpg)
> > Warped cloth, i have changed the target cloth
> 
> How did you change the target cloth?

You can change it in the test_pairs txt file

**Comment by cv38** [comment]
> Hmm, its strange. Looks like other results are also coming out as unexpected, e.g., im_h, composite map etc.
> 
> Did you test with the original pairs? Is it also producing unexpected results?
> 
> Can you please check the opencv, pillow versions as well?
> 
> It's hard to get just by the results. If you can debug the sample to see the intermediate results, that would be great.

I retrained gmm and tom model, the results look good now. but they are very blurr.

![000501_0](https://user-images.githubusercontent.com/66823829/89611344-94b9f700-d89a-11ea-8e1e-3ef9d96de9f4.jpg)

**Comment by cv38** [comment]
@minar09 , can you please suggest how we can fix the blurr image

**Comment by thaithanhtuan** [comment]
![image](https://user-images.githubusercontent.com/19486377/89873320-3459eb00-dbf5-11ea-92b0-f930824cf9a3.png)
It looks like something wrong with the input of the network.
The extracted head and bottom cloth is getting from input model image. So, it should be clear, but in your case, it is blur. Can you check these inputs?

**Comment by cv38** [comment]
> ![image](https://user-images.githubusercontent.com/19486377/89873320-3459eb00-dbf5-11ea-92b0-f930824cf9a3.png)
> It looks like something wrong with the input of the network.
> The extracted head and bottom cloth is getting from input model image. So, it should be clear, but in your case, it is blur. Can you check these inputs?

I have removed the normalization part of transforms, as it was throwing some error, do you think it will affect?

**Comment by minar09** [comment]
@cv38 , yes normalization is important. For the error part, I think either you need to install PyTorch 0.4.1 or update your transform function as discussed in https://github.com/minar09/cp-vton-plus/issues/6 or https://github.com/minar09/cp-vton-plus/issues/3#issuecomment-654727125

**Comment by thaithanhtuan** [comment]
> > ![image](https://user-images.githubusercontent.com/19486377/89873320-3459eb00-dbf5-11ea-92b0-f930824cf9a3.png)
> > It looks like something wrong with the input of the network.
> > The extracted head and bottom cloth is getting from input model image. So, it should be clear, but in your case, it is blur. Can you check these inputs?
> 
> I have removed the normalization part of transforms, as it was throwing some error, do you think it will affect?

If you want to change the input format data, you have to retrain the network with that input data format.

**Comment by cv38** [comment]
thank you @minar09  and @thaithanhtuan
